packages/shared/src/hooks/useChat.ts

```diff
@@ -130,26 +133,28 @@ export const useChat = ({ id: idFromProps }: UseChatProps): UseChat => {
   );
 
   const executePrompt = useCallback(
-    (value: string) => {
+    async (value: string) => {
       if (!value) {
         return;
       }
 
-      if (endStreamRef.current) {
-        endStreamRef.current();
+      if (sourceRef.current?.OPEN) {
+        sourceRef.current.close();
       }
 
       setSearchQuery(undefined);
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      endStreamRef.current = mockChatStream({ onMessage });
+      const source = await sendSearchQuery(value, accessToken.token);
```

Comment:
Why do you pass it, was wondering if it's not easier to abstract it into the sendSearchQuery?

Reply:
Hmm. The only concern is, we are only trying to add callbacks for now. Though if there will be anything specific to Search, then I think we can have it inside the `sendSearchQuery`. Wdyt?

Reply:
Not sure I understand, this function is the only one that needs specific token right?
it's dedicated to search already.

Reply:
Oh, my bad. Misunderstood. You mean the token? I thought it would be based on the user's JWT token. Which is coming from the AuthContext. So the function won't be able to use the hook as it is a plain function.

Reply:
ah yep, you are right 👍